CSharp/DevVmPowershell/Helpers/Implementations/ProcessingHelper.cs

```diff
@@ -782,117 +427,385 @@ public async Task<bool> RemoveWorkerManagerServerFromDefaultResourcePoolAsync()
+			{
+				throw new System.Exception("Failed to Delete Worker Manager Server");
+			}
+		}
 
-			// Setup for checking if Worker Server exists
-			Relativity.Services.TextCondition conditionWorker = new Relativity.Services.TextCondition()
+		private async Task<int> GetWorkerServerArtifactIdAsync(HttpClient httpClient)
+		{
+			int workerServerArtifactId = -1;
+			var resourceServerQueryPayload = new
 			{
-				Field = Constants.Processing.NameField,
-				Operator = Relativity.Services.TextConditionEnum.EqualTo,
-				Value = Constants.Processing.ResourceServerName
+				Query = new
+				{
+					Condition = ""
+				}
 			};
+			string serverQuery = JsonConvert.SerializeObject(resourceServerQueryPayload);
+			HttpResponseMessage queryServerResponse = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourceServer.QueryEndpointUrl, serverQuery);
+			if (!queryServerResponse.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to query for Worker Resource Server");
+			}
+			string queryServerResultString = await queryServerResponse.Content.ReadAsStringAsync();
+			dynamic queryServerResult = JObject.Parse(queryServerResultString) as JObject;
+			if (Convert.ToInt32(queryServerResult.TotalCount) > 0)
+			{
+				foreach (dynamic obj in queryServerResult.Results)
+				{
+					if (obj["Artifact"]["ServerType"]["Name"].ToString() == "Worker")
+					{
+						workerServerArtifactId = Convert.ToInt32(obj["Artifact"]["ArtifactID"].ToString());
+						break;
+					}
+				}
+			}
 
-			Relativity.Services.Query queryWorker = new Relativity.Services.Query()
+			return workerServerArtifactId;
+		}
+
+		private async Task EnableProcessingOnWorkerServerAsync(HttpClient httpClient, int workerServerId)
+		{
+			var enableProcessingOnWorker = new
 			{
-				Condition = conditionWorker.ToQueryString()
+				workerArtifactId = workerServerId
 			};
+			string request = JsonConvert.SerializeObject(enableProcessingOnWorker);
+			HttpResponseMessage response = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourceServer.EnableProcessingOnWorkerEndpointUrl, request);
+			if (!response.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to Enable Processing on Worker Server");
+			}
+		}
 
+		private async Task UpdateCategoriesOnWorkerForProcessingAsync(HttpClient httpClient, int workerServerId)
+		{
+			var updateCategories = new
+			{
+				workerArtifactId = workerServerId,
+				categories = new string[]
+				{
+					"NativeImaging",
+					"BasicImaging",
+					"SaveAsPDF",
+					"Processing"
+				}
+			};
+			string request = JsonConvert.SerializeObject(updateCategories);
+			HttpResponseMessage response = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourceServer.UpdateCategoriesOnWorkerEndpointUrl, request);
+			if (!response.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to Update Categories on Worker Server");

```

Comment:
Is there additional details in the HttpResponse regarding the actual error. We should probably put that in the Exception we are throwing which will be helpful when troubleshooting.

Reply:
Added failure message